uberfire-extensions/uberfire-metadata/uberfire-metadata-commons-io/src/main/java/org/uberfire/ext/metadata/io/index/MetadataIndexEngine.java

```diff
@@ -72,7 +72,7 @@ public boolean freshIndex(KCluster cluster) {
     @Override
     public boolean isIndexReady(KCluster cluster, String indexerId) {
         final MultiIndexerLock lock;
-        return !provider.isFreshIndex(cluster) && (lock = batchLocks.get(cluster)) != null && !lock.isLockedBy(indexerId);
+        return !provider.isFreshIndex(cluster) && ((lock = batchLocks.get(cluster)) == null || !lock.isLockedBy(indexerId));
```

Comment:
May be this method should have a test, wdyt?

Reply:
Yeah, this could use a test. I will update soon.